pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -455,8 +464,8 @@ func (az *Cloud) cleanupLoadBalancer(clusterName string, service *v1.Service, is
 	return nil
 }
 
-func (az *Cloud) ensurePublicIPExists(serviceName, pipName string) (*network.PublicIPAddress, error) {
-	pip, existsPip, err := az.getPublicIPAddress(pipName)
+func (az *Cloud) ensurePublicIPExists(serviceName string, pipResourceGroup string, pipName string) (*network.PublicIPAddress, error) {
+	pip, existsPip, err := az.getPublicIPAddress(pipResourceGroup, pipName)
```

Comment:
With the current design, does this need to handle the case where `pipResourceGroup` is the empty string, or is there something else that guarantees that will never happen?

Reply:
Now I changed it accept a service object instead and get the right resource group inside this method.